src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -488,7 +488,7 @@ bool usesOriginalInput(BoundDecisionDagNode node)
                 bool canShareInputs,
                 out BoundExpression savedInputExpression)
             {
-                int count = loweredInput.Arguments.Length;
+                int count = loweredInput.Type.TupleElements.Length;
```

Comment:
nit: consider extracting `loweredInput.Type.TupleElements` as a local (also used on line 498), then remove the `count` local.